### ISSUE TITLE
success/failure blocks not executing

### ISSUE THREAD

**Issue by kb** [activeadmin/inherited_resources#177]
If I have an action similar to the one below... The failure block appears to never be evaluated and defaults to a 422 status code.

```
class FoobarController < InheritedResources::Base
  respond_to: :html, :json
  def create
    create! do |success, failure|
      success.any do
        render :text => "foo"
      end

     failure.any do
       render :text => "bar", :status => 200
     end
  end
end
```

The resulting error code is a 422 coming from this method within actionpack... lib/action_controller/metal/responder.rb

```
def display_errors
  controller.render format => resource.errors, :status => :unprocessable_entity
end
```

Based on the rdoc of the Responder class, a 422 will be returned by default.

```
Then the response status should be 200 Created  # cucumber-api-steps-c60f156fb0bf/lib/cucumber/api_steps.rb:27
 expected: 200
 got: 422 (using ==) (RSpec::Expectations::ExpectationNotMetError)
 features/crash.feature:131:in `Then the response status should be 200 Created'
```

This seems to indicate that the block.call within the respond_with_dual_blocks for an object where .errors is not empty isn't evaluating correctly.

**Comment by kb** [comment]
I tried to duplicate this on a small generic application and everything seems to work. Therefore the issue must be contained to this the specific application I'm seeing the issue on. I'm going to close this issue as I don't believe its something within the Inherited Resources code base at the moment.

**Comment by lmarburger** [comment]
Out of curiosity, did you ever track down the cause of this issue? I happened across this exact behavior during a Rails 3.0.9 to 3.1.4 upgrade.

**Comment by kb** [comment]
Yes. If I remember correctly. I believe I just patched it in my application code and never submitted a patch back to rails.

I will try to find the code for you.

**Comment by kb** [comment]
I just quickly glanced through the issues on the rails repo and it appears a fix for this is in 3.2, however if you are using 3.1.x try patching rails with the following code.

```
module ActionController #:nodoc:
  class Responder
    # All other formats follow the procedure below. First we try to render a
    # template, if the template is not available, we verify if the resource
    # responds to :to_format and display it.
    #
    def to_format
      if get? || !has_errors? || !@default_response.nil?
        default_render
      else
        display_errors
      end
    rescue ActionView::MissingTemplate => e
      api_behavior(e)
    end
  end
end
```

**Comment by lmarburger** [comment]
Thanks, @kb. [I created a similar monkeypatch](https://gist.github.com/2782870) but I didn't look closely enough to see that this was changed in 3.2. Good to know.
